lightning/src/offers/parse.rs

```diff
@@ -58,7 +58,7 @@ mod sealed {
 
 			let parsed = CheckedHrpstring::new::<NoChecksum>(encoded.as_ref())?;
 			let hrp = parsed.hrp();
-			if hrp.as_str() != Self::BECH32_HRP {
+			if (hrp.as_str() != Self::BECH32_HRP) && (hrp.as_str().to_lowercase() != Self::BECH32_HRP) {
```

Comment:
This also allows mixed case. Not sure what the intention. If only pure-uppercase is to allowed, maybe this would be more appropriate:

```
if (hrp.as_str() != Self::BECH32_HRP) && (hrp.as_str() != Self::BECH32_HRP.to_uppercase()) {
```

Reply:
Could we use `Hrp::lowercase_char_iter` to avoid allocating a `String`? Then I think we can compare against `Self::BECH32_HRP.chars()`.